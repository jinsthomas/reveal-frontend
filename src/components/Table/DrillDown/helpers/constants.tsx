/** Constants */
import React from 'react';

export const ID = 'id';
export type ID = typeof ID;

export const PARENT_ID = 'parent_id';
export type PARENT_ID = typeof PARENT_ID;

export const ROOT_PARENT_ID = null;
export type ROOT_PARENT_ID = typeof ROOT_PARENT_ID;

export const LINKER_ITEM_CSS_CLASS = 'dd-linker-item';
export type LINKER_ITEM_CSS_CLASS = typeof LINKER_ITEM_CSS_CLASS;

export const CLICKABLE_CSS_CLASS = 'dd-linker-item dd-clickable';
export type CLICKABLE_CSS_CLASS = typeof CLICKABLE_CSS_CLASS;

export const CARET_CSS_CLASS = 'dd-caret';
export type CARET_CSS_CLASS = typeof CARET_CSS_CLASS;

export const CARET_SPAN = <span className={CARET_CSS_CLASS}>&nbsp;&#9660;</span>;
export type CARET_SPAN = typeof CARET_SPAN;

export const DEFAULT_ROW_HEIGHT = '2em';
export type DEFAULT_ROW_HEIGHT = typeof DEFAULT_ROW_HEIGHT;
