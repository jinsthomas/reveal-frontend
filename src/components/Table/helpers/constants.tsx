/** Constants */
import React from 'react';
export const ID = 'id';
export type ID = typeof ID;

export const PARENT_ID = 'parent_id';
export type PARENT_ID = typeof PARENT_ID;

export const ROOT_PARENT_ID = null;
export type ROOT_PARENT_ID = typeof ROOT_PARENT_ID;

export const LINKER_ITEM_CSS_CLASS = 'dd-linker-item';
export type LINKER_ITEM_CSS_CLASS = typeof LINKER_ITEM_CSS_CLASS;

export const CLICKABLE_CSS_CLASS = 'dd-linker-item dd-clickable';
export type CLICKABLE_CSS_CLASS = typeof CLICKABLE_CSS_CLASS;

export const CARET_CSS_CLASS = 'dd-caret';
export type CARET_CSS_CLASS = typeof CARET_CSS_CLASS;

export const CARET = <span className={CARET_CSS_CLASS}>&nbsp;&#9660;</span>;
export type CARET = typeof CARET;

export const ROWS_TO_DISPLAY = 'Rows to display';
export type ROWS_TO_DISPLAY = typeof ROWS_TO_DISPLAY;

export const PAGE = 'page';
export type PAGE = typeof PAGE;

export const OF = 'of';
export type OF = typeof OF;
