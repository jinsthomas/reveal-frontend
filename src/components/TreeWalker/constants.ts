export const ACTIVE = 'Active';
export const FIND_BY_ID = 'location/findByJurisdictionIds';
export const FIND_BY_PROPERTIES = 'location/findByProperties';
export const LOCATION = 'location';
export const COULDNT_LOAD_PARENTS = 'Could not load parents';
