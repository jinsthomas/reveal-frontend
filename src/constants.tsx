// env vars
export const WEBSITE_NAME = process.env.REACT_APP_WEBSITE_NAME;
export type WEBSITE_NAME = typeof WEBSITE_NAME;

// strings
export const LOCATION = 'Location';
export type LOCATION = typeof LOCATION;
export const PROVINCE = 'Province';
export type PROVINCE = typeof PROVINCE;
export const HOME = 'Home';
export type HOME = typeof HOME;
export const FOCUS_INVESTIGATION = 'Focus Investigation';
export type FOCUS_INVESTIGATION = typeof FOCUS_INVESTIGATION;
export const FOCUS_INVESTIGATIONS = 'Focus Investigations';
export type FOCUS_INVESTIGATIONS = typeof FOCUS_INVESTIGATIONS;

// internal urls
export const HOME_URL = '/';
export type HOME_URL = typeof HOME_URL;
export const IRS_URL = '/irs';
export type IRS_URL = typeof IRS_URL;
export const FI_URL = '/focus-investigation';
export type FI_URL = typeof FI_URL;
export const FI_HISTORICAL_URL = '/focus-investigation/historical';
export type FI_HISTORICAL_URL = typeof FI_HISTORICAL_URL;

// colors
export const GREEN = 'Green';
export type GREEN = typeof GREEN;
export const YELLOW = 'Yellow';
export type YELLOW = typeof YELLOW;
export const ORANGE = 'Orange';
export type ORANGE = typeof ORANGE;
export const RED = 'Red';
export type RED = typeof RED;

// thresholds
export const GREEN_THRESHOLD = 0.9;
export type GREEN_THRESHOLD = typeof GREEN_THRESHOLD;
export const YELLOW_THRESHOLD = 0.2;
export type YELLOW_THRESHOLD = typeof YELLOW_THRESHOLD;
export const ORANGE_THRESHOLD = 0.8;
export type ORANGE_THRESHOLD = typeof ORANGE_THRESHOLD;

// 1-3-7 thresholds
export const THRESHOLD_137 = 0;
