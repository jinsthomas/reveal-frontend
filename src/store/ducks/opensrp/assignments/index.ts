/** Assignments redux module */
import { get } from 'lodash';
import { Store } from 'redux';
import { AnyAction } from 'redux';
import SeamlessImmutable from 'seamless-immutable';

/** The reducer name */
export const reducerName = 'assignments';

/** interface for a Assignment object */
export interface Assignment {
  jurisdiction: string;
  organization: string;
  plan: string;
  fromDate: string;
  toDate: string;
}

// action interfaces

/** action type for action that adds Assignments to store */
export const ASSIGNMENTS_FETCHED = 'src/store/ducks/assignments/reducer/ASSIGNMENTS_FETCHED';
/** action type for REMOVE_TEAMS action */
export const REMOVE_ASSIGNMENTS = 'src/store/ducks/assignments/reducer/REMOVE_ASSIGNMENTS';

/** interface for Assignments fetched action */
interface FetchAssignmentsAction extends AnyAction {
  assignmentsByPlanId: { [key: string]: Assignment[] };
  type: typeof ASSIGNMENTS_FETCHED;
}

/** interface for action that removes assignments from store */
interface RemoveAssignmentsAction extends AnyAction {
  assignmentsByPlanId: { [key: string]: Assignment[] };
  type: typeof REMOVE_ASSIGNMENTS;
}

/** single type for all action types */
type AssignmentActionTypes = FetchAssignmentsAction | RemoveAssignmentsAction | AnyAction;

/** interface for Assignments state in store */
interface AssignmentsStoreState {
  assignmentsByPlanId: { [key: string]: Assignment[] };
}

// immutable assignments state in dux
export type ImmutableAssignmentsStoreState = AssignmentsStoreState &
  SeamlessImmutable.ImmutableObject<AssignmentsStoreState>;

/** initial state for Assignments records in store */
const initialOrgsStoreState: ImmutableAssignmentsStoreState = SeamlessImmutable({
  assignmentsByPlanId: {},
});

/** the Assignment reducer function */
export default function reducer(state = initialOrgsStoreState, action: AssignmentActionTypes) {
  switch (action.type) {
    case ASSIGNMENTS_FETCHED:
      return SeamlessImmutable({
        ...state,
        assignmentsByPlanId: {
          ...state.assignmentsByPlanId,
          ...action.assignmentsByPlanId,
        },
      });
    case REMOVE_ASSIGNMENTS:
      return SeamlessImmutable({
        ...state,
        assignmentsByPlanId: action.assignmentsByPlanId,
      });
    default:
      return state;
  }
}

/** action to remove assignments form store */
export const removeAssignmentsAction: RemoveAssignmentsAction = {
  assignmentsByPlanId: {},
  type: REMOVE_ASSIGNMENTS,
};

// action creators

/** creates action to add fetched assignments to store
 * @param {Assignment []} assignmentsList - array of assignments to be added to store
 *
 * @returns {FetchAssignmentsAction} - action with assignments payload that is added to store
 */
export const FetchAssignments = (assignmentsList: Assignment[]): FetchAssignmentsAction => {
  const assignmentsByPlanId: { [key: string]: Assignment[] } = {};

  for (const assignment of assignmentsList) {
    if (!assignmentsByPlanId[assignment.plan]) {
      assignmentsByPlanId[assignment.plan] = [];
    }
    assignmentsByPlanId[assignment.plan].push(assignment);
  }

  return {
    assignmentsByPlanId,
    type: ASSIGNMENTS_FETCHED,
  };
};

// selectors

/** get assignments as an object where their ids are the keys and the objects
 * the values
 * @param {Partial<Store>} state - Portion of the store
 *
 * @return {[key: string]: Assignment}
 */
export function getAssignmentsByPlanId(state: Partial<Store>): { [key: string]: Assignment[] } {
  return (state as any)[reducerName].assignmentsByPlanId;
}

/** Get all assignments as an array
 * @param {Partial<Store>} state - Part of the redux store
 * @param {string} planId - The id of the plan of which to return associated Assignments
 *
 * @return {Assignment []} - all assignments by planId in store as an array
 */
export function getAssignmentsArrayByPlanId(state: Partial<Store>, planId: string): Assignment[] {
  return get(getAssignmentsByPlanId(state), planId);
}
