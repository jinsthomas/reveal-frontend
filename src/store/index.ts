import { combine } from '@onaio/redux-reducer-registry';
import reducerRegistry from '@onaio/redux-reducer-registry';
import { connectRouter, routerMiddleware } from 'connected-react-router';
import { createBrowserHistory } from 'history';
import { applyMiddleware, compose, createStore, Reducer } from 'redux';
import thunk from 'redux-thunk';
import users, { reducerName as usersReducer } from './ducks/users';

/** Add redux devtools to Window */
declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
  }
}

/** Declare history variable */
export const history = createBrowserHistory();

/** Declare type for reducer Registry */
export interface Registry {
  [key: string]: Reducer;
}

/** Function to create the store */
export function getStore(reducers: Registry) {
  /** Register each of the initial reducers */
  Object.keys(reducers).forEach(reducerName => {
    reducerRegistry.register(reducerName, reducers[reducerName]);
  });

  /** Combine reducers */
  const reducer = combine(reducerRegistry.getReducers());

  /** Add redux devtools to enhancers */
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

  /** Create the store */
  return createStore(reducer, composeEnhancers(applyMiddleware(thunk, routerMiddleware(history))));
}

/** Router reducer */
const connectReducer = connectRouter(history);

/** Initial reducers in the reducer registry */
const defaultReducers: Registry = {
  router: connectReducer as any /** Dirty hack because connectRouter LocationChangeAction does not extend Redux.AnyAction */,
};

/** Add users reducer to registry */
defaultReducers[usersReducer] = users;

/** The initial store for the reveal web app */
const store = getStore(defaultReducers);

/** Set listener to add reducers to store when registered */
reducerRegistry.setChangeListener(reducers => {
  store.replaceReducer(combine(reducers));
});

export default store;
