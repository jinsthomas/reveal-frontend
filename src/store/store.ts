import { combine } from '@onaio/redux-reducer-registry';
import reducerRegistry from '@onaio/redux-reducer-registry';
import { connectRouter, routerMiddleware } from 'connected-react-router';
import { createBrowserHistory } from 'history';
import { applyMiddleware, createStore, Reducer } from 'redux';
import thunk from 'redux-thunk';
import users, { reducerName as usersReducer } from './ducks/users';

export const history = createBrowserHistory();

/** Declare type for reducer Registry */
export interface Registry {
  [key: string]: Reducer;
}

/** Function to create the store */
export function getStore(reducers: Registry) {
  /** Register each of the initial reducers */
  Object.keys(reducers).forEach(reducerName => {
    reducerRegistry.register(reducerName, reducers[reducerName]);
  });
  /** Combine reducers */
  const reducer = combine(reducerRegistry.getReducers());
  /** Create the store */
  return createStore(
    reducer,
    (window as any).__REDUX_DEVTOOLS_EXTENSION__ && (window as any).__REDUX_DEVTOOLS_EXTENSION__(),
    applyMiddleware(thunk, routerMiddleware(history))
  );
}

/** Router reducer */
const connectReducer = connectRouter(history);
/** Initial reducers in the reducer registry */
const defaultReducers: Registry = {
  router: connectReducer as any /** Dirty hack because connectRouter LocationChangeAction does not extend Redux.AnyAction */,
};
// /** Add users reducer to registry */
defaultReducers[usersReducer] = users;

/** The initial store for the reveal web app */
const store = getStore(defaultReducers);

/** Set listener to add reducers to store when registered */
reducerRegistry.setChangeListener(reducers => {
  store.replaceReducer(combine(reducers));
});

export default store;
