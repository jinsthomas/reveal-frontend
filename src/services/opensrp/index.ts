import { OpenSRPService as OpenSRPServiceWeb } from '@opensrp/server-service';
import { IncomingHttpHeaders } from 'http';
import { OPENSRP_API_BASE_URL } from '../../configs/env';
import store from '../../store';
import { getAccessToken } from '../../store/selectors';

/** allowed http methods */
type HTTPMethod = 'GET' | 'POST' | 'PUT' | 'DELETE';

/** get default HTTP headers for OpenSRP service */
export function getDefaultHeaders(
  accept: string = 'application/json',
  authorizationType: string = 'Bearer',
  contentType: string = 'application/json;charset=UTF-8'
): IncomingHttpHeaders {
  return {
    accept,
    authorization: `${authorizationType} ${getAccessToken(store.getState())}`,
    'content-type': contentType,
  };
}

/** interface to describe URL params object */
export interface URLParams {
  [key: string]: string | number | boolean;
}

/** converts filter params object to string
 * @param {URLParams} obj - the object representing filter params
 * @returns {string} filter params as a string
 */
export function getFilterParams(obj: URLParams | {}): string {
  return Object.entries(obj)
    .map(([key, val]) => `${key}:${val}`)
    .join(',');
}

/** get payload for fetch
 * @param {AbortSignal} signal - signal object that allows you to communicate with a DOM request
 * @param {HTTPMethod} method - the HTTP method
 * @returns the payload
 */

export function getPayloadOptions(_: AbortSignal, method: HTTPMethod) {
  return {
    headers: getDefaultHeaders() as HeadersInit,
    method,
  };
}

/** The OpenSRP service class
 * Extends class OpenSRPService from OpenSRPService web
 * Sample usage:
 * -------------
 * const service = new OpenSRPService('the-endpoint');
 *
 * **To list all objects**: service.list()
 *
 * **To get one object**: service.read('the-object-identifier')
 *
 * **To create a new object**: service.create(theObject)
 *
 * **To update an object**: service.update(theObject)
 */
export class OpenSRPService extends OpenSRPServiceWeb {
  constructor(
    endpoint: string,
    baseURL: string = OPENSRP_API_BASE_URL,
    getPayload: typeof getPayloadOptions = getPayloadOptions
  ) {
    super(baseURL, endpoint, getPayload);
  }
  /** create method
   * Send a POST request to the general endpoint containing the new object data
   * Successful requests will result in a HTTP status 201 response with no body
   * @param {T} data - the data to be POSTed
   * @param {params} params - the url params object
   * @param {HTTPMethod} method - the HTTP method
   * @returns the object returned by API
   */
  public async create<T>(data: T, params: any = null, method: HTTPMethod = 'POST'): Promise<{}> {
    const url = OpenSRPService.getURL(this.generalURL, params);
    const payload = {
      ...this.getOptions(this.signal, method),
      'Cache-Control': 'no-cache',
      Pragma: 'no-cache',
      body: JSON.stringify(data),
    };
    const response = await fetch(url, payload);

    if (!response.ok || response.status !== 201) {
      throw new Error(
        `OpenSRPService create on ${this.endpoint} failed, HTTP status ${response.status}`
      );
    }

    return {};
  }
}
