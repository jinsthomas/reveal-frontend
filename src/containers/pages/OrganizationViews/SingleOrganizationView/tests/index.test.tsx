import ListView from '@onaio/list-view';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import Helmet from 'react-helmet';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { SINGLE_TEAM_URL, TEAM } from '../../../../../constants';
import store from '../../../../../store';
import * as organizationDucks from '../../../../../store/ducks/opensrp/organizations';
import * as practitionersDucks from '../../../../../store/ducks/opensrp/practitioners';
import * as fixtures from '../../../../../store/ducks/tests/fixtures';
import ConnectedSingleOrgView, { SingleOrganizationView } from '../../SingleOrganizationView';

// tslint:disable-next-line: no-var-requires
const fetch = require('jest-fetch-mock');

const history = createBrowserHistory();

describe('src/containers/pages/TeamAssignment', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders SingleTeamView without crashing', () => {
    // clean test: check for conditions resulting in errors
    const mock: any = jest.fn();
    const props = {
      history,
      location: mock,
      match: {
        isExact: true,
        params: { id: '' },
        path: `${SINGLE_TEAM_URL}/:id`,
        url: `${SINGLE_TEAM_URL}/teamId`,
      },
      organization: fixtures.organization1,
      practitioners: fixtures.practitioners,
    };

    shallow(
      <Router history={history}>
        <SingleOrganizationView {...props} />
      </Router>
    );
  });

  it('renders SingleTeamView correctly', () => {
    // clean test: check for stuff that can only be present if page loaded
    fetch.once(JSON.stringify([])).once(JSON.stringify([]));
    const mock: any = jest.fn();
    const props = {
      history,
      location: mock,
      match: {
        isExact: true,
        params: { id: '' },
        path: `${SINGLE_TEAM_URL}/:id`,
        url: `${SINGLE_TEAM_URL}/teamId`,
      },
      organization: fixtures.organization1,
      practitioners: [fixtures.practitioner1],
    };

    const wrapper = mount(
      <Router history={history}>
        <SingleOrganizationView {...props} />
      </Router>
    );

    // check for page title is set
    const helmet = Helmet.peek();
    expect(helmet.title).toEqual(`${TEAM} - ${fixtures.organization1.name}`);

    // check for breadcrumbs is set
    expect(toJson(wrapper.find('HeaderBreadcrumb'))).toMatchSnapshot('HeaderBreadcrumb');

    // text wise : check for the team Name text - should be in the top-most tier header
    const pageTitle = wrapper.find('.page-title');
    expect(pageTitle.length).toEqual(1);
    expect(pageTitle.text()).toEqual(`${fixtures.organization1.name}`);

    // check: team details div(snapshot)
    expect(toJson(wrapper.find('#team-details'))).toMatchSnapshot('TeamDetails');

    // check: team members div(snapshot)
    expect(wrapper.find(ListView).length).toEqual(1);
    wrapper.unmount();
  });

  it('Plays well with the store', async () => {
    // it gets data from store correctly
    store.dispatch(organizationDucks.fetchOrganizations([fixtures.organization3]));
    store.dispatch(
      practitionersDucks.fetchPractitionerRoles(
        fixtures.org3Practitioners,
        fixtures.organization3.identifier
      )
    );

    fetch
      .once(JSON.stringify([fixtures.organization3]))
      .once(JSON.stringify(fixtures.org3Practitioners));
    const mock: any = jest.fn();
    const props = {
      history,
      location: mock,
      match: {
        isExact: true,
        params: { id: `${fixtures.organization3.identifier}` },
        path: `${SINGLE_TEAM_URL}/:id`,
        url: `${SINGLE_TEAM_URL}/1`,
      },
    };
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedSingleOrgView {...props} />
        </Router>
      </Provider>
    );

    await new Promise(resolve => setImmediate(resolve));

    const passedProps = wrapper.find(SingleOrganizationView).props() as any;
    expect(passedProps.team).toEqual(fixtures.organization1);
    expect(passedProps.teamMembers).toEqual([fixtures.practitioner1]);
  });

  it('calls api correctly', async () => {
    // api calls are correct.
    const mock: any = jest.fn();
    const mockRead: any = jest.fn(async () => []);
    const mockList: any = jest.fn(async () => []);
    const serviceMock: any = jest.fn(() => {
      return {
        list: mockList,
        read: mockRead,
      };
    });

    const props = {
      history,
      location: mock,
      match: {
        isExact: true,
        params: { id: `${fixtures.organization3.identifier}` },
        path: `${SINGLE_TEAM_URL}/:id`,
        url: `${SINGLE_TEAM_URL}/1`,
      },
    };
    mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedSingleOrgView {...props} />
        </Router>
      </Provider>
    );
    await new Promise(resolve => setImmediate(resolve));

    expect(serviceMock).toHaveBeenCalledTimes(2);
    expect(mockRead.mock.calls).toHaveBeenCalledWith(fixtures.organization3.identifier);
    // expect(mockList.mock.calls).toHaveBeenCalledWith();
    expect.assertions(3);
  });

  xit('Deleting api calls for removing a practitioner from an organization', async () => {
    // removing a practitioner from an organization workflow
    fetch
      .once(JSON.stringify(fixtures.organization3))
      .once(JSON.stringify(fixtures.org3Practitioners));
    const mock: any = jest.fn();
    const mockDelete: any = jest.fn(async () => []);
    const serviceMock: any = jest.fn(() => {
      return {
        delete: mockDelete,
      };
    });

    const props = {
      history,
      location: mock,
      match: {
        isExact: true,
        params: { id: '1' },
        path: `${SINGLE_TEAM_URL}/:id`,
        url: `${SINGLE_TEAM_URL}/1`,
      },
    };
    mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedSingleOrgView {...props} />
        </Router>
      </Provider>
    );
    await new Promise(resolve => setImmediate(resolve));
    // simulate removal action

    expect(serviceMock).toHaveBeenCalledTimes(3);
    expect(mockDelete.mock.calls).toHaveBeenCalledWith();
    expect.assertions(3);
  });
});

describe('src/../singleOrganization.selectors', () => {
  it('selectors are called with the correct arguments', async () => {
    // calls selectors correctly
    const organizationByIdMock = jest.spyOn(organizationDucks, 'getOrganizationById');
    const practitionerByOrgIdMock = jest.spyOn(practitionersDucks, 'getPractitionersByOrgId');

    fetch
      .once(JSON.stringify(fixtures.organization3))
      .once(JSON.stringify(fixtures.org3Practitioners));

    const mock: any = jest.fn();
    const props = {
      history,
      location: mock,
      match: {
        isExact: true,
        params: { id: `${fixtures.organization3.identifier}` },
        path: `${SINGLE_TEAM_URL}/:id`,
        url: `${SINGLE_TEAM_URL}/1`,
      },
    };
    mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedSingleOrgView {...props} />
        </Router>
      </Provider>
    );

    await new Promise(resolve => setImmediate(resolve));
    expect(organizationByIdMock.mock.calls).toEqual([
      selectorState,
      fixtures.organization3.identifier,
    ]);
    expect(practitionerByOrgIdMock.mock.calls).toEqual([
      selectorState,
      fixtures.organization3.identifier,
    ]);
  });
});
