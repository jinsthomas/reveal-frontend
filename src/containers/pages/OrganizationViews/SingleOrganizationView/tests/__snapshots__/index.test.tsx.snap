// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/containers/pages/TeamAssignment renders SingleTeamView correctly: HeaderBreadcrumb 1`] = `
<HeaderBreadcrumb
  currentPage={
    Object {
      "label": "fcc19470-d599-11e9-bb65-2a2ae2dbcce4",
      "url": "/teams/view/fcc19470-d599-11e9-bb65-2a2ae2dbcce4",
    }
  }
  pages={
    Array [
      Object {
        "label": "Home",
        "url": "/",
      },
      Object {
        "label": "Teams",
        "url": "/teams",
      },
    ]
  }
>
  <div>
    <Breadcrumb
      aria-label="breadcrumb"
      className="reveal-breadcrumb"
      listTag="ol"
      tag="nav"
    >
      <nav
        aria-label="breadcrumb"
        className="reveal-breadcrumb"
      >
        <ol
          className="breadcrumb"
        >
          <BreadcrumbItem
            key="0"
            tag="li"
          >
            <li
              className="breadcrumb-item"
            >
              <Link
                key="0"
                replace={false}
                to="/"
              >
                <a
                  href="/"
                  onClick={[Function]}
                >
                  Home
                </a>
              </Link>
            </li>
          </BreadcrumbItem>
          <BreadcrumbItem
            key="1"
            tag="li"
          >
            <li
              className="breadcrumb-item"
            >
              <Link
                key="1"
                replace={false}
                to="/teams"
              >
                <a
                  href="/teams"
                  onClick={[Function]}
                >
                  Teams
                </a>
              </Link>
            </li>
          </BreadcrumbItem>
          <BreadcrumbItem
            active={true}
            tag="li"
          >
            <li
              aria-current="page"
              className="active breadcrumb-item"
            >
              fcc19470-d599-11e9-bb65-2a2ae2dbcce4
            </li>
          </BreadcrumbItem>
        </ol>
      </nav>
    </Breadcrumb>
  </div>
</HeaderBreadcrumb>
`;

exports[`src/containers/pages/TeamAssignment renders SingleTeamView correctly: TeamDetails 1`] = `null`;
